community/kernel/src/main/java/org/neo4j/kernel/impl/api/OperationsFacade.java

```diff
@@ -397,6 +401,10 @@ public Object relationshipGetProperty( long relationshipId, int propertyKeyId )
         {
             return relationship.get().getProperty( propertyKeyId );
         }
+        finally
+        {
+            statement.assertOpen();
+        }
```

Comment:
Why do we have this check after we've read the property value? Property values aren't lazy. Same for `nodeGetProperty` above.